library/Icinga/Authentication/IcingaTotp.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Icinga\Authentication;
+
+use chillerlan\QRCode\QRCode;
+use Icinga\Common\Database;
+use Icinga\Exception\ConfigurationError;
+use Icinga\Model\TotpModel;
+use ipl\Sql\Connection;
+use ipl\Sql\Delete;
+use ipl\Sql\Insert;
+use ipl\Stdlib\Filter;
+use OTPHP\TOTP;
+use Throwable;
+
+class IcingaTotp
+{
+    use Database;
+
+    protected TOTP $totp;
+
+    protected string $user;
+
+    private function __construct(string $user)
+    {
+        $this->user = $user;
+    }
+
+    protected function setTotp(TOTP $totp): static
+    {
+        $this->totp = $totp;
+
+        return $this;
+    }
+
+    public static function generate(string $user): static
```

Comment:
> We will use the defaults of 6 digit tokens, 30 seconds interval and sha-1 algorithm for secret generation.

The default? So it's not strictly required? Good! Because SHA1 is insecure af and both Authy and Google support more than that.

(Bitte gleich gescheit machen.)